src/systems/terrain_quad_tree.zig

```diff
@@ -87,15 +87,16 @@ const Mesh = struct {
     num_vertices: u32,
 };
 
+const invalid_index = std.math.maxInt(u32);
 const QuadTreeNode = struct {
     center: [2]f32,
     size: [2]f32,
     child_indices: [4]u32,
     mesh_lod: u32,
     patch_index: [2]u32,
     // TODO: Do not store these here when we implement streaming
-    heightmap_handle: gfx.TextureHandle,
-    splatmap_handle: gfx.TextureHandle,
+    heightmap_handle: ?gfx.TextureHandle,
+    splatmap_handle: ?gfx.TextureHandle,
 
     pub inline fn containsPoint(self: *QuadTreeNode, point: [2]f32) bool {
```

Comment:
Just realized - the floats here are guaranteed to be at integer boundaries right? I'm just a little bit concerned with floating point inprecision here, like is there a chance a point will be inbetween two nodes and neither will think it's inside them? If so would it be better to store the left-bottom corner instead of the center?

Reply:
Btw for me a more intuitive way to write this would be to use "to the right of left AND to the left of right AND..etc" rather than "Not to the left of left...". But the result should be the same :)

Reply:
oh yeah, I think doing it with `AND`s will also short-circuit :) I'll update it, especially if it's more intuitive.

Regarding the floating point precision issue and being on the edge, I totally get it. How would it be different though if we store the bottom left? (genuine question)

Reply:
Hehe well I'm not sure if it would actually matter - it's why I asked about the integer boundaries. The corners are guaranteed to be there right,  but the center I wasn't sure about (especially if the patches are n^2+1)

Reply:
In the case of meshes, patches dimensions are always powers of 2 (currenlty 64, 126, 256 and 512), so also their centers are guaranteed to be at integer boundaries (32, 64, 128 and 256).

Reply:
Ok! Then we're all good :)